packages/remirror__extension-link/src/link-extension.ts

```diff
@@ -290,7 +290,16 @@ export class LinkExtension extends MarkExtension<LinkOptions> {
             }
 
             const href = node.getAttribute('href');
-            const auto = node.hasAttribute(AUTO_ATTRIBUTE);
+            const text = node.textContent;
+
+            // If link text content equals href value we "auto link"
+            // e.g [test](//test.com) - not "auto link"
+            // e.g [test.com](//test.com) - "auto link"
+            const auto =
+              node.hasAttribute(AUTO_ATTRIBUTE) ||
+              href === text ||
+              href?.replace(`${this.options.defaultProtocol}//`, '') === text;
```

Comment:
Think we should also check `this.options.autoLink` is enabled for the new checks here?

Reply:
If the autoLink option is not set to true does it matter if that value is true or false?

Should I change this to 

```ts 
const auto =
this.options.autoLink &&
(node.hasAttribute(AUTO_ATTRIBUTE) ||
href === text ||
href?.replace(`${this.options.defaultProtocol}//`, "") ===
text);
```

Reply:
> If the autoLink option is not set to true does it matter if that value is true or false?

I just want to avoid the potential confusion of people seeing `auto: true` in there doc state when the think they have auto linking disabled.

Your suggested code makes sense to me

Reply:
Thank you for the explanation.

Pushing the change.